### PR TITLE
Add TTF Fonts and Simple UI

### PR DESCRIPTION
**Pull request by SeedyROM**
Merging in the basic changes for simple UI. Added the `UIObject` and `UITest` structs to test out simple UI functionality.

### CODE REVIEW

**Comment by SeedyROM** [comment]
Fixed a merge conflict from the `Rasterizer.cpp` that was caused by a lack of `SDL_RenderPresent` in my branch.

File: Source/Rasterizer.cpp [hunk 1]
```diff
@@ -124,8 +124,9 @@ void Rasterizer::render(SDL_Renderer* renderer, int sizeFactor = 1) {
 	SDL_Rect destinationRect = { 0, 0, sizeFactor * width, sizeFactor * height };
 
 	SDL_UpdateTexture(screenTexture, NULL, pixelBuffer, width * sizeof(Uint32));
-	SDL_RenderCopy(renderer, screenTexture, NULL, &destinationRect);
-	SDL_RenderPresent(renderer);
+
+	SDL_RenderCopy(renderer, screenTexture, NULL, NULL);
```
Comment (mbellman) [inline Source/Rasterizer.cpp]: We'll want to keep `&destinationRect` as the fourth argument to `SDL_RenderCopy`. It's used for the "pixel filter" scaling mechanism.

File: Source/main.cpp [hunk 2]
```diff
@@ -5,9 +5,17 @@ int width = 1200;
...
 int main(int argc, char* argv[]) {
-	Engine engine(width, height, DEBUG_DRAWTIME);
+	Engine engine(width, height);
 
-	Mesh mesh(100, 40, 50);
+	TTF_Font* mono = TTF_OpenFont("./Assets/FreeMono.ttf", 15);
+
+	UIText text("Test Text");
+
+	text.setRenderer(engine.getRenderer());
```
Comment (mbellman) [inline Source/main.cpp]: For ease of use, and to make `Engine`'s implementation a little more opaque, I'd move this step into `addUIObject` and allow `UIText` (or other `UIObject`s) to be created, configured, and added freely without needing to explicitly acquire the renderer.

```
void Engine::addUIObject(UIObject* uiObject) {
  uiObject.setRenderer(renderer);

  ui->addObject(uiObject);
}
```

**Review by mbellman — COMMENTED** [review]
Thanks a bunch for taking care of this. This will prove invaluable in freely composing HUDs/menus/what have you. I left two comments - one required change and one thing I'd like to be changed. ; >

File: Source/Rasterizer.cpp (hunk 1 above)
Comment (SeedyROM) [reply Source/Rasterizer.cpp]: Can do, not sure how I accidentally removed that.

File: Source/main.cpp (hunk 2 above)
Comment (SeedyROM) [reply Source/main.cpp]: Fixed!

**Comment by SeedyROM** [comment]
Fixed everything up!

**Review by mbellman — APPROVED** [review]

**Pull request merged**